infra/README.md

```diff
@@ -0,0 +1,55 @@
+# DeltaNYC Infrastructure
+
+Infrastructure is maintained using;
+
+* [Terraform](https://www.terraform.io/)
+* [Cloud-Init](http://cloudinit.readthedocs.io/en/latest/index.html)
+
+
+# To the droplet's ip address
+
+```bash
+$ curl -s -X GET -H "Content-Type: application/json" -H "Authorization: Bearer $DIGITALOCEAN_TOKEN" "https://api.digitalocean.com/v2/droplets" | jq '.[] | .[] | select(.name=="delta_lookup_tool") | .networks.v4 | .[]'
+```
+
+We're currently using the DO ubuntu image, which allows ssh directly to the root account.
+
+```bash
+ssh -i ~/.ssh/id_rsa_delta root@{public_id}
+```
+
+The key to the machine is maintained by Chris Henry
+
+
+# Configuration
+
+### Google Maps API Key.
+
+The key in use was generated from Chris Henry's Google API Console, but any key
+with Google Maps permissions can be used.
+
+# Setup steps
+
+We should be using luigid, but I can't quite figure out what it's supposed to be
+doing.
```

Comment:
Not sure why there'd be a 137 exit code - they should exit successfully. My guess is there's an error of some sorts in one of the tasks? Each worker should log the general status of tasks, plus logs from the task (the import script) itself; if these aren't logging to stdout, they can be examined with `docker-compose logs luigi-import-worker` + grepping for the particular worker in question. Luigid also doubles as a web server; accessing it in a web browser shows a page with the status of all tasks. It can save a complete history of executed tasks, but that requires some extra setup I haven't finished yet.

While it may be possible to specify multiple tasks for a worker, our setup runs the worker with a single task. That task, however, is simply a `WrapperTask` that lists all the import scripts as deps. (This is ImportAll in luigi_master) As mentioned, the worker will start up, list out all the imports and check if they've been completed. Luigi then submits the list of tasks to the scheduler, which will send back a task if one exists that is not already in progress. Once the worker finishes (with either success or failure), it tells the scheduler the result and gets a new task (if one exists that is not already in progress). This means that if you're running multiple workers, the exact number of tasks each worker will run is not set in stone, and will depend on timing. Of course, if you run a single worker, the number of tasks that worker is determined, since that worker has to do all the work.